lib/invoice_item_repository.rb

```diff
@@ -0,0 +1,58 @@
+require 'csv'
+
+class InvoiceItemRepository
+  def initialize(path)
+    @path = path
+    @rows = CSV.read(@path, headers: true, header_converters: :symbol)
+    @all = all
+  end
+
+  def all
+    result = @rows.map do |row|
+      InvoiceItem.new(row)
+    end
+  end
+
+  def find_by_id(id)
+    result = @all.find do |row|
+      row.id == id
+    end
+    result
+  end
+
+  def find_all_by_item_id(item_id)
+    result = @all.find_all do |row|
+      row.item_id == item_id
+    end
+    result
```

Comment:
similar to above, I think the `result` on lines 24 and 27 is unnecessary; `#find_all` should return the collection by itself